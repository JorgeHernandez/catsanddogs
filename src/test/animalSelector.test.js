import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import AnimalSelector from '../components/animal_selector';

it('::::autocheck::::', () => {
  	expect(1-1).toBe(0);
});

it('renders a snapshot', () => {
  	const tree = renderer.create(<AnimalSelector onSearchTermChange={()=>{} } />).toJSON();
  	expect(tree).toMatchSnapshot();
});

it('renders an AnimalSelector component', () => {
  const component = shallow(<AnimalSelector />);
  expect(component).toHaveLength(1);
});

it('initial state animal should be empty', () => {
  const wrapper = shallow(<AnimalSelector onSearchTermChange={()=>{} }/>); 
  expect(wrapper.state().animal).toBe("");
});

it('cats option changes state animal to cats', () => {
	const wrapper = shallow(<AnimalSelector onSearchTermChange={()=>{} }/>); 
	wrapper.instance().onInputChange('cats');
	expect(wrapper.state().animal).toBe('cats');
});

it('dogs option changes state animal to dogs', () => {
	const wrapper = shallow(<AnimalSelector onSearchTermChange={()=>{} }/>); 
	wrapper.instance().onInputChange('dogs');
	expect(wrapper.state().animal).toBe('dogs');
});

it('cats option changes state animal to cats', () => {
    const wrapper = shallow(<AnimalSelector onSearchTermChange={()=>{} }/>); 
    wrapper.instance().onInputChange('cats');
    expect(wrapper.state().animal).toBe('cats');
});
