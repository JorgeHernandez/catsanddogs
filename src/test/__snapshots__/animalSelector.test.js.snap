// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimalSelector changes state animal to cats when cat radio is selected 1`] = `
<div
  className="center-on-page"
>
  <input
    checked={false}
    id="cats"
    name="animal"
    onChange={[Function]}
    type="radio"
    value="cats"
  />
  <label>
    Cats
  </label>
  <input
    checked={false}
    id="dogs"
    name="animal"
    onChange={[Function]}
    type="radio"
    value="dogs"
  />
  <label>
    Dogs
  </label>
</div>
`;

exports[`AnimalSelector changes state animal to cats when cat radio is selected 2`] = `
<div
  className="center-on-page"
>
  <input
    checked={false}
    id="cats"
    name="animal"
    onChange={[Function]}
    type="radio"
    value="cats"
  />
  <label>
    Cats
  </label>
  <input
    checked={false}
    id="dogs"
    name="animal"
    onChange={[Function]}
    type="radio"
    value="dogs"
  />
  <label>
    Dogs
  </label>
</div>
`;

exports[`renders a snapshot 1`] = `
<div
  className="center-on-page"
>
  <input
    checked={false}
    id="cats"
    name="animal"
    onChange={[Function]}
    type="radio"
    value="cats"
  />
  <label>
    Cats
  </label>
  <input
    checked={false}
    id="dogs"
    name="animal"
    onChange={[Function]}
    type="radio"
    value="dogs"
  />
  <label>
    Dogs
  </label>
</div>
`;
